xmtp_db/src/encrypted_store/consent_record.rs

```diff
@@ -74,7 +74,7 @@ impl StoredConsentRecord {
                 last_consent.state,
                 hex::encode(&group.id),
             );
-            cr.store(conn)?;
+            conn.insert_newer_consent_record(cr)?;
```

Comment:
Important change, cc @nplasterer and @humanagent:
- This function is called when processing the Welcome for a DM. Previously, if the consent sync inserted the consent entry before the Welcome for the DM was processed, the call to `cr.store(conn)?;` would fail with a unique constraint error, because we're trying to insert a consent record that already exists. This would abort processing of the Welcome and lead to a DM sub-group that was never received, causing the appearance of a fork.
- Now, we check if the consent record already exists, and either update it or no-op without returning an error, so that the Welcome is not dropped.